pkg/deployments/src/artifact.ts

```diff
@@ -48,11 +63,16 @@ export function checkArtifact(task: Task): void {
 /**
  * Read the build-info file for the contract `contractName` and extract the ABI and bytecode.
  */
-function extractContractArtifact(task: Task, contractName: string): CompilerOutputContract {
-  const buildInfo = task.buildInfo(contractName);
+function extractContractArtifact(task: Task, fileName: string, contractName: string): CompilerOutputContract {
+  const buildInfo = task.buildInfo(fileName);
 
   // Read ABI and bytecode from build-info file.
-  const contractSourceName = findContractSourceName(buildInfo, contractName);
+  let contractSourceName: string;
+  try {
+    contractSourceName = findContractSourceName(buildInfo, contractName);
+  } catch (e) {
+    throw new Error(`Could not find a unique source for ${contractName} in the file ${fileName}`);
+  }
```

Comment:
We already raise similar errors inside `findContractSourceName` and this message would likely end up being confusing in the case where the contract isn't found (which is the more likely error) as it implies that there's more than one.

Reply:
Thanks; I was a bit in doubt about this change too.